### PR TITLE
Forward the pagination headers in the GitLab response

### PR DESCRIPTION
**Pull request by ableuler**
Forward pagination headers in the response, modify URLs in the `Link` header such that they point to a gateway URL.

### CODE REVIEW

File: app/processors/gitlab_processor.py [hunk 1]
```diff
@@ -45,8 +56,30 @@ def urlencode_paths(path):
...
+            app.config['GITLAB_URL'], app.config['HOST_NAME']
+        )
+
+    return headers
+
+
 class GitlabGeneric(BaseProcessor):
 
+    def __init__(self, path, endpoint):
+        self.path = path
```
Comment (ebiiii) [inline app/processors/gitlab_processor.py]: why not `super().__init__(path, endpoint)`

File: app/processors/gitlab_processor.py [hunk 2]
```diff
@@ -56,6 +89,17 @@ def process(self, request, headers):
         self.endpoint = urljoin(self.endpoint.format(**app.config), self.path)
         return super().process(request, headers)
 
+    def create_response_headers(self, response):
+
+        headers = Headers()
+        for header in response.headers:
+            if header in self.forwarded_headers:
+                headers.add(header, response.headers[header])
```
Comment (ebiiii) [inline app/processors/gitlab_processor.py]: here too

File: app/processors/gitlab_processor.py (hunk 1 above)
Comment (ableuler) [reply app/processors/gitlab_processor.py]: yep, you're right!

File: app/processors/gitlab_processor.py (hunk 2 above)
Comment (ableuler) [reply app/processors/gitlab_processor.py]: here too ;)

**Pull request merged**